tests/e2e/transaction/transactionSubmission.test.ts

```diff
@@ -619,6 +633,57 @@ describe("transaction submission", () => {
       });
     });
   });
+  describe("Multi Key", () => {
+    test("it submits a multi key transaction", async () => {
+      const multiKey = new MultiKey({
+        publicKeys: [
+          singleSignerED25519SenderAccount.publicKey,
+          legacyED25519SenderAccount.publicKey,
+          singleSignerSecp256k1Account.publicKey,
+        ],
+        signaturesRequired: 2,
+      });
+      const des = new Deserializer(multiKey.bcsToBytes());
+      expect(multiKey).toEqual(MultiKey.deserialize(des));
+
+      const authKey = AuthenticationKey.fromPublicKey({ publicKey: multiKey });
+
+      const multiKeyAccountAddress = authKey.derivedAddress();
+
+      await aptos.fundAccount({ accountAddress: multiKeyAccountAddress.toString(), amount: 100_000_000 });
+
+      const transaction = await aptos.generateTransaction({
+        sender: multiKeyAccountAddress.toString(),
+        data: {
+          function: `0x${contractPublisherAccount.accountAddress.toStringWithoutPrefix()}::transfer::transfer`,
+          functionArguments: [new U64(1), receiverAccounts[0].accountAddress],
+        },
+      });
+      // create a bitmap where singleSignerED25519SenderAccount and singleSignerSecp256k1Account
+      const bitmap = multiKey.createBitmap({ bits: [0, 2] });
+
+      // account1 and account3 sign the transaction
+      const account1Authenticator = aptos.signTransaction({ signer: singleSignerED25519SenderAccount, transaction });
+      const account3Authenticator = aptos.signTransaction({ signer: singleSignerSecp256k1Account, transaction });
+
+      const multiKeyAuth = new AccountAuthenticatorMultiKey(
+        multiKey,
+        [
+          // TODO find a fix
```

Comment:
We can probably just make a common function on all types of authenticators, or do something that will extract it no matter what.  Mixing classes and interfaces is messy.

Let's look at this in a couple days

Reply:
an optional solution that needs to look at is function overload. `signTransaction` return `AccountAuthenticator` type, and if one wants to access inner variables, it needs to tell ts what is the type they want to use.